packages/python/plotly/plotly/express/_core.py

```diff
@@ -1575,6 +1554,26 @@ def build_dataframe(args, constructor):
         value_name = _escape_col_name(columns, "value", [])
         var_name = _escape_col_name(columns, var_name, [])
 
+    if isinstance(args['data_frame'], nw.DataFrame) and needs_interchanging:
+        # Interchange to PyArrow
+        if wide_mode:
+            args["data_frame"] = nw.from_native(args['data_frame'].to_arrow(), eager_only=True)
+        else:
+            # Save precious resources by only interchanging columns that are
+            # actually going to be plotted. This is tricky to do in the general case,
+            # because Plotly allows calls like `px.line(df, x='x', y=['y1', df['y1']])`,
+            # but interchange-only objects (e.g. DuckDB) don't typically have a concept
+            # of self-standing Series. It's more important to perform project pushdown
+            # here seeing as we're materialising to an (eager) PyArrow table.
+            necessary_columns = {
+                i for i in args.values() if isinstance(i, str) and i in columns
+            }
+            for field in args:
+                if args[field] is not None and field in array_attrables:
+                    necessary_columns.update(i for i in args[field] if i in columns)
+            columns = list(necessary_columns)
+            args["data_frame"] = nw.from_native(args['data_frame'].select(columns).to_arrow(), eager_only=True)
+
```

Comment:
Should we set

```suggestion
        import pyarrow as pa
        native_namespace = pa
```
?

Reply:
good one, thanks!